src/NServiceBus.Core/Pipeline/PipelineModelBuilder.cs

```diff
@@ -45,8 +60,17 @@ public List<RegisterStep> Build()
                 throw new Exception(message);
             }
 
-            //  Step 2: do replacements
-            foreach (var metadata in replacements)
+            //  Step 2: validate and apply replacements
+            var groupedReplacements = replacements.GroupBy(x => x.ReplaceId);
+            if (groupedReplacements.Any(x => x.Count() > 1))
```

Comment:
Can Count() be .Any()?

Reply:
If you mean:
if (groupedReplacements.Any(x => x.Any()))
then yes, but is it more readable? I'm on the fence

Reply:
Looking at the code, I take back what I said. ´if (groupedReplacements.Any(x => x.Any()))´ wouldn't work, because what we're testing here, is to verify if there are multiple replacements. So we're not checking if there are `Any`, but actually "more than one".

Reply:
Sorry, I thought it was `>=1` my bad!